src/Http/Routing/src/Matching/HttpMethodMatcherPolicy.cs

```diff
@@ -396,9 +396,18 @@ private Endpoint CreateRejectionEndpoint(IEnumerable<string> httpMethods)
 
         private static bool ContainsHttpMethod(List<string> httpMethods, string httpMethod)
         {
-            for (var i = 0; i < httpMethods.Count; i++)
+            var methods = httpMethods.ToArray();
```

Comment:
@benaadams @rynowak  Is allocating here helps the performance? I want to make sure that I got the point correctly.

Reply:
That seems backwards. Normally you'd want to avoid allocations.

Reply:
Yeah, this feels weird to me. I think I misunderstand the point of this https://github.com/dotnet/aspnetcore/pull/21277#discussion_r417489933 .
What should I do here?

Reply:
Oh, I think he meant for you to change the method signature to take in an array, but that doesn't seem plausible because the list is mutable. You can try the AsSpan trick, but if that doesn't work then revert this.

Reply:
`var methods = CollectionsMarshal.AsSpan(httpMethods)` creates a span over the List's underlying array, bounded by the List's length. [CollectionsMarshal.cs#L18-L19](https://github.com/dotnet/runtime/blob/1a959590ea1e21be367ac78f1f4d3b85e0679c92/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/CollectionsMarshal.cs#L18-L19)

Then the Span should be very fast to loop over as it will eliminate bounds checks etc

Reply:
```suggestion
            var methods = CollectionsMarshal.AsSpan(httpMethods)
```

Reply:
@benaadams Do you suggest every for loop on list in this repo should use `CollectionsMarshal.AsSpan`?

Reply:
Not necessarily; if it's in a hot path might be worth checking if it improves things